elytron/src/main/java/org/wildfly/extension/elytron/AbstractCredentialStoreResourceDefinition.java

```diff
@@ -184,6 +237,109 @@ protected void removeAliasOperation(OperationContext context, ModelNode operatio
+            ModelNode algorithmModel = ALGORITHM.resolveModelAttribute(context, operation);
+            if (algorithmModel.isDefined()) {
+                algorithm = algorithmModel.asString();
+            } else {
+                algorithm = RSA_ALGORITHM;
+            }
+
+            int size;
+            ModelNode sizeModel = SIZE.resolveModelAttribute(context, operation);
+            if (sizeModel.isDefined()) {
+                size = sizeModel.asInt();
+            } else {
+                size = 0;
+            }
+
+            KeyPairCredential keyPairCredential = new KeyPairCredential(generateKeyPair(algorithm, size));
+            storeKeyPair(credentialStore, alias, keyPairCredential);
+        } catch (GeneralSecurityException e) {
+            throw ROOT_LOGGER.keyPairOperationFailed(ElytronDescriptionConstants.GENERATE_KEY_PAIR, dumpCause(e), e);
+        }
+    }
+
+    protected void exportKeyPairPublicKeyOperation(OperationContext context, ModelNode operation, CredentialStore credentialStore) throws OperationFailedException {
+        try {
+            String alias = ALIAS.resolveModelAttribute(context, operation).asString();
+
+            KeyPairCredential credential = credentialStore.retrieve(alias, KeyPairCredential.class);
+            if (credential == null) {
+                throw ROOT_LOGGER.credentialDoesNotExist(alias, KeyPairCredential.class.getSimpleName());
+            }
+
+            PublicKey publicKey = credential.getKeyPair().getPublic();
+            String exportedPublicKey = exportPublicKey(publicKey);
+
+            ModelNode result = context.getResult();
+            result.get(ElytronDescriptionConstants.PUBLIC_KEY).set(exportedPublicKey);
+        } catch (GeneralSecurityException e) {
+            throw ROOT_LOGGER.keyPairOperationFailed(ElytronDescriptionConstants.EXPORT_KEY_PAIR_PUBLIC_KEY, dumpCause(e), e);
+        }
+    }
+
+    protected void importKeyPairOperation(OperationContext context, ModelNode operation, CredentialStore credentialStore) throws OperationFailedException {
+        try {
+            String privateKeyContent;
+            String publicKeyContent = null;
+
+            String alias = ALIAS.resolveModelAttribute(context, operation).asString();
+
+            if (credentialStore.exists(alias, KeyPairCredential.class)) {
+                throw ROOT_LOGGER.credentialAlreadyExists(alias, KeyPairCredential.class.getName());
+            }
+
+            ModelNode privateKeyString = PRIVATE_KEY_STRING.resolveModelAttribute(context, operation);
+            ModelNode publicKeyString = PUBLIC_KEY_STRING.resolveModelAttribute(context, operation);
+            ModelNode privateKeyLocation = PRIVATE_KEY_LOCATION.resolveModelAttribute(context, operation);
+            ModelNode publicKeyLocation = PUBLIC_KEY_LOCATION.resolveModelAttribute(context, operation);
+
+            String passphrase = PASSPHRASE.resolveModelAttribute(context, operation).asString();
```

Comment:
Looks like the `PASSPHRASE` is optional so it could be null here. Should it be required?

Reply:
In general, would be good to double check which attributes are required for each operation.

Reply:
> Looks like the `PASSPHRASE` is optional so it could be null here. Should it be required?

Passphrase can be null if none was used to encrypt the key, while it is not recommended. I have added a test case to check it.